strawberry/ext/mypy_plugin.py

```diff
@@ -12,11 +12,21 @@ def lazy_type_analyze_callback(ctx: AnalyzeTypeContext) -> Type:
     return type_
 
 
+def private_type_analyze_callback(ctx: AnalyzeTypeContext) -> Type:
+    type_name = ctx.type.args[0]
+    type_ = ctx.api.analyze_type(type_name)
+
+    return type_
+
+
 class StrawberryPlugin(Plugin):
     def get_type_analyze_hook(self, fullname: str):
         if fullname == "strawberry.lazy_type.LazyType":
             return lazy_type_analyze_callback
 
+        if fullname == "strawberry.private.Private":
+            return private_type_analyze_callback
+
```

Comment:
hero! I'm glad this wasn't too difficult 😊

Reply:
can we add a test for this here: https://github.com/strawberry-graphql/strawberry/blob/master/tests/mypy/test_plugin.yml?

Reply:
Done